code401challenges/src/main/java/code401challenges/utilities/AnimalShelter.java

```diff
@@ -0,0 +1,42 @@
+package code401challenges.utilities;
+
+import code401challenges.stacksandqueues.Queue;
+
+public class AnimalShelter {
+
+    Queue<Animal> dogQ = new Queue<>();
+    Queue<Animal> catQ = new Queue<>();
+
+
+    public String enqueue(Animal animal) {
+        String message = "";
+        if(animal.type == "dog") {
+            dogQ.enqueue(animal);
+            message = "dog is added successfully";
+        } else if(animal.type == "cat") {
+            catQ.enqueue(animal);
+            message = "cat is added successfully";
+        } else {
+            message = "it has to be either dog or cat";
+        }
+        return message;
+    }
+
+    public String dequeue(Animal animal) {
```

Comment:
This method is supposed to return the dog or cat that was dequeue, not a string!